### PR TITLE
Performance Tweaks

### PR DESCRIPTION
**Pull request by elgorro**
@schana We talked as 1.10 was released or somewhat - now had the time to realize. As it is my first commit on DayZ-Modding, so please double check and hope u can check it with your testing-abilities.
I'm not sure about every of my solutions is supported on the engine - let me know and I will think about ;)

If u have questions, let me know :)

### CODE REVIEW

File: SchanaModParty/Game/schana_party_marker_manager_server.c [hunk 1]
```diff
@@ -36,9 +36,9 @@ class SchanaPartyMarkerManagerServer {
 
     void ServerRegisterMarkers (string id, ref array<ref SchanaPartyMarkerInfo> playerMarkers) {
         int maxMarkers = GetSchanaPartyServerSettings ().GetMaxMarkers ();
-        if (maxMarkers != -1) {
-            while (playerMarkers.Count () > maxMarkers) {
-                playerMarkers.RemoveOrdered (0);
+        if (maxMarkers > -1 && playerMarkers.Count () > maxMarkers ) {
+            for (int i = maxMarkers; playerMarkers.Count (); ++maxMarkers) {
+                playerMarkers.Remove (i);
```
Comment (elgorro) [inline SchanaModParty/Game/schana_party_marker_manager_server.c:41]: I'm not sure whats the correct command here (remove or delete?)

**Comment by DaemonF0rge** [comment]
So with part of this fix you set the 	 `map<string, DayZPlayer> id_map;` to be a global map, the issue with doing this if a player logs off and the server then removes the player data it can cause an access violation when the data is attempted to be referenced

**Comment by elgorro** [comment]
@daemonforge Are try-catch-blocks working? Didn't found infos on the documentation (https://community.bistudio.com/wiki/DayZ:Enforce_Script_Syntax)

**Comment by DaemonF0rge** [comment]
@elgorro no I am about 99% sure they are not possible, to be fair never tried but also have never seen them used by anyone within the DayZ modding community

**Comment by elgorro** [comment]
thx for the answer - very unfortunate to hear. glad to hear u fixed some stuff - closing so far.

**Pull request closed without merging**